src/python/pants/backend/jvm/tasks/jvm_task.py

```diff
@@ -23,13 +26,17 @@ class JvmTask(Task):
 
   @classmethod
   def subsystem_dependencies(cls):
-    return super().subsystem_dependencies() + (JVM.scoped(cls),)
+    return super().subsystem_dependencies() + (JVM.scoped(cls), DistributionLocator, JvmPlatform)
 
   @classmethod
   def register_options(cls, register):
     super().register_options(register)
     register('--confs', type=list, default=['default'],
              help='Use only these Ivy configurations of external deps.')
+    register('--strict-jvm-version', type=bool, advanced=True, fingerprint=True,
```

Comment:
Should this maybe move to the `JVM` subsystem? We're headed in the direction of being more subsystem centric in v2, and I think that the scoping of the `JVM` subsystem should allow for the same kind of per-Task configurability of this flag, in a more forwards compatible way.

The downside of moving it there would be needing to deprecate the flag on `junit`, but that might be worth it?

Reply:
Oh: I think that this would also avoid needing to extend JvmTask in `prep_command`.

Reply:
Hm. I think I should push this back down. I think it's primarily used by tests to allow for more strict splitting of platform test shards, and while it could have use in other tasks, I'm not sure it's necessary for this set of changes.

I have other thoughts about it but they aren't quite backed yet.

Reply:
One thing I've thought is that maybe platform should have min/max defined for themselves, then strictness would be determined by the platform--that might be nice if for example, you have a code base with pre and post Java 9 code, where some pre-9 code doesn't work on post-9 version of the jvm. In that case, you'd want to enforce that those targets don't run on 9+ and fail if 9+ were unavailable.

Reply:
> One thing I've thought is that maybe platform should have min/max defined for themselves, then strictness would be determined by the platform--that might be nice if for example, you have a code base with pre and post Java 9 code, where some pre-9 code doesn't work on post-9 version of the jvm.

That sounds very reasonable.